flask_nameko/proxies.py

```diff
@@ -31,37 +38,47 @@ def configure(self, config):
     def _get_nameko_connection(self):
         proxy = ClusterRpcProxy(
             self._config,
-            timeout=self._config.get('RPC_TIMEOUT', None)
+            timeout=float(self._config.get('RPC_TIMEOUT')) if self._config.get('RPC_TIMEOUT') else None,
+            context_data=self.context_data,
+            worker_ctx_cls=self.worker_cls
         )
         return proxy.start()
 
     def get_connection(self):
         if not self._pool:
-            raise ClusterNotConfiguredError("Please configure your cluster beore requesting a connection.")
+            raise ClusterNotConfiguredError("Please configure your cluster before requesting a connection.")
         return self._pool.get_connection()
 
     def release_connection(self, connection):
         return self._pool.release_connection(connection)
 
+
 class LazyServiceProxy(object):
     def __init__(self, get_connection, service):
         self.get_connection = get_connection
         self.service = service
+
     def __getattr__(self, name):
         return getattr(getattr(self.get_connection(), self.service), name)
 
+
 class FlaskPooledClusterRpcProxy(PooledClusterRpcProxy):
-    def __init__(self, app=None, connect_on_method_call=True):
+    def __init__(self, app=None, context_data=None, worker_cls=None, connect_on_method_call=True):
+        super().__init__()
         self._connect_on_method_call = connect_on_method_call
         if app:
-            self.init_app(app)
+            self.init_app(app, context_data, worker_cls)
 
-    def init_app(self, app):
+    def init_app(self, app, context_data=None, worker_cls=None):
         config = dict()
-        for key, val in app.config.iteritems():
+        for key, val in app.config.items():
```

Comment:
It looks unimportant: 
- this code use on application start only and shouldn't decrease performance
- config dictionary is not so big
- more one site-package dependency: **six**
If you disagree, I can add, what you think?